nntrainer/layers/conv2d_layer.cpp

```diff
@@ -237,11 +247,19 @@ void Conv2DLayer::calcDerivative(sharedConstTensors derivatives) {
 
   for (unsigned int b = 0; b < in_dim.batch(); ++b) {
     Tensor inSub = derivative.getBatchSlice(b, 1);
+    std::vector<float> in_col;
```

Comment:
ditto

Reply:
I don't think it should be. This is nothing releate with tensor or even does not have dimension. If we have to we have set dimensions as (1,1,1, filter_kertnel.getDim().gerFeatureLen()*out_dim.width()*dout_dim.height() ) and I don't think any consistency here.

Reply:
The reason to use tensor is to be able to use a common allocator and manage memory in a unified way.
Further, with datatypes introduced later, vector<float> at all places will have to replaced.

Reply:
I think it does confused as well. No reason to use tensor here.

Reply:
Overhead will be minor but later, we want to manage memory ourselves. Vector does not allow managing memory ourselves. IMO all the data in nntrainer must use Tensor be it temporary or not.